### PR TITLE
Updates github workflows to follow other projects

### PR DESCRIPTION
**Pull request by nvollmar**
also disable publish to fix build until we're read to publish artifacts

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 11
+
+      - name: Cache Coursier cache
+        uses: coursier/cache-action@v6.4.0
+
+      - name: Create all API docs for artifacts/website and all reference docs
+        run: sbt "unidoc; docs/paradox"
```
Comment (mdedetrich) [inline .github/workflows/checks.yml:73]: nit: New line

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        with:
+          distribution: temurin
+          java-version: ${{ matrix.jdk }}
+
+      - name: Cache Coursier cache
+        uses: coursier/cache-action@v6.4.0
+
+      - name: Test against ${{ matrix.container }}
+        run: |-
+          docker-compose up -d ${{ matrix.container }} && sbt ${{ matrix.scalaVersion }} ${{matrix.test}}
```
Comment (mdedetrich) [inline .github/workflows/unit-tests.yml:55]: nit: New line

**Review by mdedetrich — CHANGES_REQUESTED** [review]
Looks good to me, just some minor things regarding new lines

File: .github/workflows/checks.yml (hunk 1 above)
Comment (nvollmar) [reply .github/workflows/checks.yml]: added

File: .github/workflows/unit-tests.yml (hunk 2 above)
Comment (nvollmar) [reply .github/workflows/unit-tests.yml]: added

**Comment by mdedetrich** [comment]
One of the tests fail, seems to be a flaky one so retrying

**Pull request merged**